File: src/codility/array/CyclicRotation.java
package codility.array;



public class CyclicRotation {

    public int[] solution(int[] A, int K) {

        if (A.length == 0) {
            return A;
        }

        K %= A.length;

        int[] ans = new int[A.length];

        K = A.length -K;

        int j =0;
        for (int i = K ;i< A.length;i++) {
            ans[j] = A[i];
            j++;
        }

        for (int i = 0;i<K;i++) {
            ans[j] = A[i];
            j++;
        }

        return ans;
    }
}

File: src/codility/kotlin/CyclicRotation.kt
package codility.kotlin


fun solution(A: IntArray, K: Int): IntArray {
    // write your code in Kotlin

    if (K == 0 || A.isEmpty()) {
        return A
    }

    val r = K % A.size

    val list = mutableListOf<Int>()

    for (i in 0 until r) {
        list.add(A[A.size  - r + i])
    }

    for (i in 0 until A.size - r) {
        list.add(A[i])
    }

    return list.toIntArray()
}